doc/source/data/getting-started.rst

```diff
@@ -1,229 +1,86 @@
-===============
+Dataset Quick Start
+===================
+
+Ray Dataset is an abstraction over a list of Ray object references to *blocks*, with APIs for distributed data loading and processing. Each block holds a set of items and can be in format of either an `Arrow table <https://arrow.apache.org/docs/python/data.html#tables>`__
+, a `Pandas DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__
+, or a Python list.
 
 In this tutorial you will learn how to:
 
 - Create and save a Ray ``Dataset``.
 - Transform a ``Dataset``.
 - Pass a ``Dataset`` to Ray tasks/actors and access the data inside.
 
-.. _ray_datasets_quick_start:
-
--------------------
-Dataset Quick Start
--------------------
-
-Ray Datasets implements Distributed `Arrow <https://arrow.apache.org/>`__.
-A Dataset consists of a list of Ray object references to *blocks*.
-Each block holds a set of items in either an `Arrow table <https://arrow.apache.org/docs/python/data.html#tables>`__
-(when creating from or transforming to tabular or tensor data), a `Pandas DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__
-(when creating from or transforming to Pandas data), or a Python list (otherwise).
-Let's start by creating a Dataset.
-
-Creating Datasets
-=================
-
 .. tip::
 
    Run ``pip install "ray[data]"`` to get started!
 
-You can get started by creating Datasets from synthetic data using ``ray.data.range()`` and ``ray.data.from_items()``.
-Datasets can hold either plain Python objects (i.e. their schema is a Python type), or Arrow records
-(in which case their schema is Arrow).
-
-.. code-block:: python
-
-    import ray
-
-    # Create a Dataset of Python objects.
-    ds = ray.data.range(10000)
-    # -> Dataset(num_blocks=200, num_rows=10000, schema=<class 'int'>)
-
-    ds.take(5)
-    # -> [0, 1, 2, 3, 4]
-
-    ds.count()
-    # -> 10000
+Creating and Saving Datasets
+----------------------------
 
-    # Create a Dataset of Arrow records.
-    ds = ray.data.from_items([{"col1": i, "col2": str(i)} for i in range(10000)])
-    # -> Dataset(num_blocks=200, num_rows=10000, schema={col1: int64, col2: string})
+You can create a Dataset from Python objects. These objects can be held inside Dataset as the plain Python objects (where the schema is a Python type), or as Arrow records (in which case their schema is Arrow).
 
-    ds.show(5)
-    # -> {'col1': 0, 'col2': '0'}
-    # -> {'col1': 1, 'col2': '1'}
-    # -> {'col1': 2, 'col2': '2'}
-    # -> {'col1': 3, 'col2': '3'}
-    # -> {'col1': 4, 'col2': '4'}
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __create_from_python_begin__`
+   :end-before: __create_from_python_end__
 
-    ds.schema()
-    # -> col1: int64
-    # -> col2: string
-
-Datasets can be created from files on local disk or remote datasources such as S3.
+Datasets can also be created from files on local disk or remote datasources such as S3.
```

Comment:
We should link out to the remote storage and file formats sections of the "Creating Datasets" feature guide here.

Reply:
Yeah, good point.